pkg/connector/connector.go

```diff
@@ -29,11 +29,18 @@ const (
 	TypeDestination
 )
 
+const (
+	ProvisionTypeAPI ProvisionType = iota
+	ProvisionTypeConfig
+)
+
 type (
 	// Type defines the connector type.
 	Type int
 	// Status defines the running status of a connector.
 	Status int
+	// ProvisionType defines provisioning type
+	ProvisionType int
```

Comment:
Thinking out loud here - how would it look like if this type was a string? Then we wouldn't need to have constants here (and in every other service package), the service itself wouldn't care what different provision types there are. Only the calling code would need to know what its provision type is (e.g. orchestrators would have the provision type `"api"`) and that it can only mutate entities with the same provision type.

This might remove some boilerplate code, it would also make the provision type readable in logs since it would be a string.